src/test/java/io/gravitee/common/utils/RxHelperTest.java

```diff
@@ -527,6 +527,21 @@ void shouldExponentiallyRetry() {
         }
     }
 
+    @Test
+    @DisplayName("Should retry and finally fail after 3 retries")
+    void shouldRetryWithMaxAttempt() {
+        try {
+            Flowable
+                .generate(emitter -> emitter.onError(new RuntimeException("will never work")))
+                .retryWhen(RxHelper.retryExponentialBackoff(1, 1, TimeUnit.SECONDS, 1, 3, t -> true))
+                .test()
+                .awaitDone(4, TimeUnit.SECONDS)
+                .assertError(err -> err.getMessage().equals("will never work"));
+        } finally {
+            RxJavaPlugins.reset();
```

Comment:
To improve test velocity, you could use TestScheduler as done in other tests to avoid waiting 4 seconds.